src/ephemeris.rs

```diff
@@ -14,28 +14,29 @@ use crate::{
     signal::{Code, Constellation, GnssSignal},
     time::GpsTime,
 };
-use std::fmt::{Display, Formatter};
+use std::fmt;
+use std::error::Error;
 
 /// Number of bytes in  the Galileo INAV message
 // TODO(jbangelo) bindgen doesn't catch this variable on linux for some reason
 pub const GAL_INAV_CONTENT_BYTE: usize = (128 + 8 - 1) / 8;
 
 /// An error indicating that the ephemeris is invalid
-#[derive(Copy, Clone, Debug)]
-pub struct InvalidEphemeris {}
+#[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Hash)]
+pub struct InvalidEphemeris(Status);
```

Comment:
Little oddity here, the ephemeris status `enum` includes a `Valid` variant. Not sure if it's preferable to have a separate enum with only error states?

Reply:
I've since split the statuses into separate enums.